sdk/extensions/Azure.Extensions.AspNetCore.Configuration.Secrets/src/ParallelSecretLoader.cs

```diff
@@ -24,17 +24,17 @@ public ParallelSecretLoader(SecretClient client)
             _tasks = new List<Task<Response<KeyVaultSecret>>>();
         }
 
-        public void AddSecretToLoad(string secretName)
+        public void AddSecretToLoad(string name, string version)
         {
-            _tasks.Add(Task.Run(() => GetSecretAsync(secretName)));
+            _tasks.Add(Task.Run(() => GetSecretAsync(name, version)));
         }
 
-        private async Task<Response<KeyVaultSecret>> GetSecretAsync(string secretName)
+        private async Task<Response<KeyVaultSecret>> GetSecretAsync(string name, string version)
         {
             await _semaphore.WaitAsync().ConfigureAwait(false);
             try
             {
-                return await _client.GetSecretAsync(secretName).ConfigureAwait(false);
+                return await _client.GetSecretAsync(name, version).ConfigureAwait(false);
```

Comment:
Wg tego widzę, że można nie przekazać nic (wtedy `null`):

https://github.com/mediporta/azure-sdk-for-net/blob/mediporta/sdk/keyvault/Azure.Security.KeyVault.Secrets/src/SecretClient.cs

A jak to się będzie odbywało w naszym przypadku? Możemy nie chcieć podawać nru wersji, ale cały łańcuch metod nie ma `string version = null.` Możesz pokazać jakiś przykład użycia?

Reply:
Jezeli w appsetingsach nie specyfikujemy wersji wtedy jest ona nullem. Tak zachowuje się aktualnie paczka w repo z którego forkujemy.

Czyli wersja jest null (pobiera na 90% sekret z najnowszą wersją / domyślną wersję) 
wersja nie jest nullem i jest pobierania po @ i wtedy system pobiera sekret w dokładnie tej wersji